### PR TITLE
Sync fork

### PR DESCRIPTION
**Pull request by andreasparteli**
In an effort to get rid of the deprecated usages of `::set-output` ([Trello ticket](https://trello.com/c/WyQXp2Vy)), this is the sync to the original repo where it has been fixed already.

I'm adding an info to the README as well on why we went for the fork (see also [slack discussion](https://komootteam.slack.com/archives/C69RFAJNB/p1699277199112959)).

### CODE REVIEW

**Comment by andreasparteli** [comment]
Test pipeline seems to fail because of this being a fork. At least there's no change to the original repo that could cause it.

File: entrypoint.sh [hunk 1]
```diff
@@ -1,43 +1,73 @@
...
+fi
+
+setOutput() {
+    echo "${1}=${2}" >> "${GITHUB_OUTPUT}"
+}
 
 current_branch=$(git rev-parse --abbrev-ref HEAD)
 
-pre_release="false" # never use pre_release since we want this to run on all branches
+pre_release="$prerelease"
```
Comment (andreasparteli) [inline entrypoint.sh:61]: already default `false` in new line 16

**Review by joejuzl — APPROVED** [review]
LGTM

**Comment by andreasparteli** [comment]
Teardown for a test PR in push-api repo worked well ([CircleCI run](https://app.circleci.com/pipelines/github/komoot/push-api/1223/workflows/246a6691-d7a6-459c-93d8-8498d53b61d5)). Could still be we have weird edge cases but normal use-case seems to be fine. So I'd merge it soon.

**Review by jtheuer — APPROVED** [review]
Did you spot anything fishy? I didn't. But it is quite a complex project compared to what we want to achieve with it.

The action include another 3rd party action dependency (seems to be responsible for commenting with bash syntax hints). If you don't need it I'd recommend to disable that step.

**Comment by andreasparteli** [comment]
I mainly checked the `entrypoint.sh` for issues as that's the executed code and didn't find anything suspicious.

You have a very good point though. There's a few third-party-dependencies in the GitHub workflows that run in this repo if we work on the code (e.g. the linter you mentioned; so not if the actual action gets executed). I'll just delete them as we indeed don't need them. That should still allow for easy updating in the future.

But overall it's indeed a big repo now for the simple use case we have. I'd still suggest to not manually maintain this in the future - at least at this point.
